tests-e2e/core/managed_environment_test.go

```diff
@@ -695,6 +695,80 @@ var _ = Describe("GitOpsDeployment Managed Environment E2E tests", func() {
+			Expect(err).To(BeNil())
+
+			managedEnv, secret := buildManagedEnvironment(apiServerURL, kubeConfigContents, true)
+
+			k8sClient, err := fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			err = k8s.Create(&secret, k8sClient)
+			Expect(err).To(BeNil())
+
+			err = k8s.Create(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+
+			gitOpsDeploymentResource := gitopsDeplFixture.BuildGitOpsDeploymentResource("my-gitops-depl",
+				"https://github.com/redhat-appstudio/managed-gitops", "resources/test-data/sample-gitops-repository/environments/overlays/dev",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+			gitOpsDeploymentResource.Spec.Destination.Environment = managedEnv.Name
+			gitOpsDeploymentResource.Spec.Destination.Namespace = fixture.GitOpsServiceE2ENamespace
+			err = k8s.Create(&gitOpsDeploymentResource, k8sClient)
+			Expect(err).To(BeNil())
+
+			By("ensuring GitOpsDeployment should have expected health and status and reconciledState")
+
+			expectedReconciledStateField := managedgitopsv1alpha1.ReconciledState{
+				Source: managedgitopsv1alpha1.GitOpsDeploymentSource{
+					RepoURL: gitOpsDeploymentResource.Spec.Source.RepoURL,
+					Path:    gitOpsDeploymentResource.Spec.Source.Path,
+				},
+				Destination: managedgitopsv1alpha1.GitOpsDeploymentDestination{
+					Name:      gitOpsDeploymentResource.Spec.Destination.Environment,
+					Namespace: gitOpsDeploymentResource.Spec.Destination.Namespace,
+				},
+			}
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+					gitopsDeplFixture.HaveReconciledState(expectedReconciledStateField)))
+
+			secretList := corev1.SecretList{}
+
+			err = k8sClient.List(context.Background(), &secretList, &client.ListOptions{Namespace: dbutil.DefaultGitOpsEngineSingleInstanceNamespace})
+			Expect(err).To(BeNil())
+
+			dbQueries, err := db.NewSharedProductionPostgresDBQueries(false)
+			Expect(err).To(BeNil())
+			defer dbQueries.CloseDatabase()
+
+			mapping := &db.APICRToDatabaseMapping{
+				APIResourceType: db.APICRToDatabaseMapping_ResourceType_GitOpsDeploymentManagedEnvironment,
+				APIResourceUID:  string(managedEnv.UID),
+				DBRelationType:  db.APICRToDatabaseMapping_DBRelationType_ManagedEnvironment,
+			}
+			err = dbQueries.GetDatabaseMappingForAPICR(context.Background(), mapping)
+			Expect(err).To(BeNil())
+
+			By("Verify whether AppProjectManagedEnv is created or not")
+			appProjectManagedEnvDB := db.AppProjectManagedEnvironment{
+				Managed_environment_id: mapping.DBRelationKey,
+			}
+
+			Eventually(func() bool {
+				err = dbQueries.GetAppProjectManagedEnvironmentByManagedEnvId(ctx, &appProjectManagedEnvDB)
+				return Expect(err).To(BeNil())
+			}, time.Second*60).Should(BeTrue())
+
```

Comment:
I think we can should also test the update case:

1) delete the GitOpsDeploymentManagedEnvironment, and remove the reference to the managed environment from the GitOpsDeployment.

Ensure that the `AppProject` CR no longer references the old ManagedEnvironment.